src/meta/server_state.cpp

```diff
@@ -2864,5 +2881,55 @@ void server_state::clear_app_envs(const app_env_rpc &env_rpc)
                    new_envs.c_str());
         });
 }
+
+namespace {
+
+bool validate_target_max_replica_count_internal(int32_t max_replica_count,
+                                                int32_t alive_node_count,
+                                                std::string &hint_message)
+{
+    if (max_replica_count > max_allowed_replica_count) {
+        hint_message = fmt::format("requested replica count({}) exceeds the max "
+                                   "allowed replica count({})",
+                                   max_replica_count,
+                                   max_allowed_replica_count);
+        return false;
+    }
+
+    if (max_replica_count < FLAGS_min_allowed_replica_count) {
+        hint_message = fmt::format("requested replica count({}) is less than the min "
+                                   "allowed replica count({})",
+                                   max_replica_count,
+                                   FLAGS_min_allowed_replica_count);
+        return false;
+    }
+
```

Comment:
you may don't need to split two `if` to  log the `hit_message`, consider the following whether more simple:
```c++
 if (max_replica_count > max_allowed_replica_count || max_replica_count < FLAGS_min_allowed_replica_count) {
        hint_message = fmt::format("requested replica count({}) must be range within [min={} max={}] ",
                                   max_replica_count,
                                   FLAGS_min_allowed_replica_count
                                   max_allowed_replica_count);
        return false;
    }
```

Reply:
Ok, I'll merge the both conditions. A range seems more specific.